samples/Common.c

```diff
@@ -723,18 +724,24 @@ STATUS sessionCleanupWait(PSampleConfiguration pSampleConfiguration)
                   5 * HUNDREDS_OF_NANOS_IN_A_SECOND);
 
         // Check if we need to re-create the signaling client on-the-fly
-        if (ATOMIC_LOAD_BOOL(&pSampleConfiguration->recreateSignalingClient)) {
-            CHK_STATUS(freeSignalingClient(&pSampleConfiguration->signalingClientHandle));
-            CHK_STATUS(createSignalingClientSync(&pSampleConfiguration->clientInfo,
-                                                 &pSampleConfiguration->channelInfo,
-                                                 &pSampleConfiguration->signalingClientCallbacks,
-                                                 pSampleConfiguration->pCredentialProvider,
-                                                 &pSampleConfiguration->signalingClientHandle));
-            CHK_STATUS(signalingClientConnectSync(pSampleConfiguration->signalingClientHandle));
-
+        if (ATOMIC_LOAD_BOOL(&pSampleConfiguration->recreateSignalingClient) &&
+            STATUS_SUCCEEDED(freeSignalingClient(&pSampleConfiguration->signalingClientHandle)) &&
+            STATUS_SUCCEEDED(createSignalingClientSync(&pSampleConfiguration->clientInfo,
+                    &pSampleConfiguration->channelInfo,
+                    &pSampleConfiguration->signalingClientCallbacks,
+                    pSampleConfiguration->pCredentialProvider,
+                    &pSampleConfiguration->signalingClientHandle))) {
             // Re-set the variable again
             ATOMIC_STORE_BOOL(&pSampleConfiguration->recreateSignalingClient, FALSE);
         }
+
+        // Check the signaling client state and connect if needed
+        if (IS_VALID_SIGNALING_CLIENT_HANDLE(pSampleConfiguration->signalingClientHandle)) {
+            CHK_STATUS(signalingClientGetCurrentState(pSampleConfiguration->signalingClientHandle, &signalingClientState));
+            if (signalingClientState == SIGNALING_CLIENT_STATE_READY) {
```

Comment:
why do we have to wait until state ready to call signalingClientConnectSync?

Reply:
in the master sample we just call signalingClientConnectSync

Reply:
Good question. In the startup, we will bail out on creation error. Here, we don't. We keep re-trying. Only when the creation is OK and we are in Ready state, then we will attempt to connect. Otherwise, we will loop in 5 seconds and retry re-creating the client first.

The reason I thought this would be useful is the following. Consider the sample application is running and suddenly we hit a network outage. This causes the signaling client to do it's thing and attempt to reconnect until timeout. This will then call the error callback which would trigger the re-create and re-connect. The network outage still being present will cause the re-create to fail and the application to gracefully terminate.

This is a continuous attempt to re-create/re-connect.